src/utils/ProfileUtils.ts

```diff
@@ -180,17 +180,23 @@ export async function populateImportedProfile(
     }
 }
 
-//TODO: Check if instead of returning null/empty strings, there's some errors that should be handled
-export async function readProfileFile(file: string) {
-    let read = '';
+export async function readProfileFile(file: string): Promise<string> {
+    let read: string | undefined;
     if (file.endsWith('.r2x')) {
         read = (await FsProvider.instance.readFile(file)).toString();
     } else if (file.endsWith('.r2z')) {
-        const result: Buffer | null = await ZipProvider.instance.readFile(file, "export.r2x");
-        if (result === null) {
-            return null;
-        }
-        read = result.toString();
+        await ZipProvider.instance.readFile(file, "export.r2x").then(
+            (value) => { read = value?.toString() },
```

Comment:
This won't compile for me (and the CI pipeline) so I can't test it. I assume you can't use the optional chaining here yet.

Reply:
@VilppeRiskidev Did something happen here during the force push? I could swear this was fixed before, but now the `?` operator is back?

Reply:
yeah, weirdly the changes were lost at some point :( I found some of them from IDE's local history. Remade the changes and should be good now.